File: server/router/workflow/workflow.go
package workflow

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type WorkflowRouter struct{}

func (r *WorkflowRouter) InitWorkflowRouter(Router *gin.RouterGroup) {
	workflowRouter := Router.Group("workflow").Use(middleware.OperationRecord())
	workflowRouterWithoutRecord := Router.Group("workflow")
	{
		workflowRouter.POST("createPipeline", workflowApi.CreatePipeline)   // 创建流水线
		workflowRouter.PUT("updatePipeline", workflowApi.UpdatePipeline)    // 更新流水线
		workflowRouter.DELETE("deletePipeline", workflowApi.DeletePipeline) // 删除流水线
		workflowRouter.POST("togglePipeline", workflowApi.TogglePipeline)   // 启用/停用
		workflowRouter.POST("clonePipeline", workflowApi.ClonePipeline)     // 克隆流水线
		workflowRouter.POST("triggerBuild", workflowApi.TriggerBuild)       // 触发构建
		workflowRouter.POST("cancelBuild", workflowApi.CancelBuild)         // 取消构建
		workflowRouter.POST("retryBuild", workflowApi.RetryBuild)           // 重跑构建
		workflowRouter.POST("approveStage", workflowApi.ApproveStage)       // 审批 gate
	}
	{
		workflowRouterWithoutRecord.POST("getPipelineList", workflowApi.GetPipelineList)   // 流水线列表
		workflowRouterWithoutRecord.GET("findPipeline", workflowApi.FindPipeline)           // 流水线详情
		workflowRouterWithoutRecord.POST("getBuildList", workflowApi.GetBuildList)          // 构建历史
		workflowRouterWithoutRecord.GET("getBuildDetail", workflowApi.GetBuildDetail)       // 构建详情
		workflowRouterWithoutRecord.GET("getBuildLog", workflowApi.GetBuildLog)             // 构建日志
		// SSE 长连接: 绝不能套 TimeoutMiddleware(参考 utils/sse/hub.go:168 约束)
		workflowRouterWithoutRecord.GET("buildStream", workflowApi.BuildStream)             // 构建实时事件
	}
}

// InitWebhookRouter 注册公开 webhook 触发路由(挂 PublicGroup, 免登录, 靠 secret 头校验)
func (r *WorkflowRouter) InitWebhookRouter(PublicRouter *gin.RouterGroup) {
	publicWebhook := PublicRouter.Group("webhook")
	{
		publicWebhook.POST("trigger/:id", workflowApi.WebhookTrigger) // webhook 触发
	}
}

File: web/src/api/workflow.js
import service from '@/utils/request'

// ============================== 流水线定义 ==============================

export const createPipeline = (data) => {
  return service({
    url: '/workflow/createPipeline',
    method: 'post',
    data
  })
}

export const updatePipeline = (data) => {
  return service({
    url: '/workflow/updatePipeline',
    method: 'put',
    data
  })
}

export const deletePipeline = (data) => {
  return service({
    url: '/workflow/deletePipeline',
    method: 'delete',
    data
  })
}

export const togglePipeline = (data) => {
  return service({
    url: '/workflow/togglePipeline',
    method: 'post',
    data
  })
}

export const clonePipeline = (data) => {
  return service({
    url: '/workflow/clonePipeline',
    method: 'post',
    data
  })
}

export const getPipelineList = (data) => {
  return service({
    url: '/workflow/getPipelineList',
    method: 'post',
    data
  })
}

export const findPipeline = (params) => {
  return service({
    url: '/workflow/findPipeline',
    method: 'get',
    params
  })
}

// ============================== 构建 ==============================

export const triggerBuild = (data) => {
  return service({
    url: '/workflow/triggerBuild',
    method: 'post',
    data
  })
}

export const cancelBuild = (data) => {
  return service({
    url: '/workflow/cancelBuild',
    method: 'post',
    data
  })
}

export const retryBuild = (data) => {
  return service({
    url: '/workflow/retryBuild',
    method: 'post',
    data
  })
}

export const approveStage = (data) => {
  return service({
    url: '/workflow/approveStage',
    method: 'post',
    data
  })
}

export const getBuildList = (data) => {
  return service({
    url: '/workflow/getBuildList',
    method: 'post',
    data
  })
}

export const getBuildDetail = (params) => {
  return service({
    url: '/workflow/getBuildDetail',
    method: 'get',
    params
  })
}

export const getBuildLog = (params) => {
  return service({
    url: '/workflow/getBuildLog',
    method: 'get',
    params
  })
}

// ============================== 工具:状态映射 ==============================

// 构建状态 -> 显示文案 / Element Plus tag 类型
export const buildStatusMap = {
  pending: { label: '待执行', type: 'info' },
  running: { label: '执行中', type: 'warning' },
  'running-approval': { label: '待审批', type: 'warning' },
  success: { label: '成功', type: 'success' },
  failed: { label: '失败', type: 'danger' },
  canceled: { label: '已取消', type: 'info' }
}

export const buildStatusLabel = (status) =>
  (buildStatusMap[status] && buildStatusMap[status].label) || status

export const buildStatusType = (status) =>
  (buildStatusMap[status] && buildStatusMap[status].type) || 'info'
